src/main/java/net/oauth/jsontoken/AbstractJsonTokenParser.java

```diff
@@ -0,0 +1,229 @@
+/**
+ * Copyright 2010 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package net.oauth.jsontoken;
+
+import com.google.common.base.Preconditions;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import net.oauth.jsontoken.crypto.AsciiStringVerifier;
+import net.oauth.jsontoken.crypto.SignatureAlgorithm;
+import net.oauth.jsontoken.crypto.Verifier;
+
+import org.apache.commons.codec.binary.Base64;
+import org.joda.time.Instant;
```

Comment:
This might be a big ask, but is it possible to make the AsyncJsonTokenParser use Java Time (available from Java 8 onwards) and just do conversions internally as necessary? There's been a big recent push to migrate from Joda time to Java time.

Reply:
Yeah, the AsyncJsonTokenParser can be implemented using Java time